### ISSUE TITLE
Use tf.math.divide_no_nan here

### ISSUE THREAD

**Issue by ehtshamelahiwm** [keras-team/keras#18152]
https://github.com/keras-team/keras/blob/f9336cc5114b4a9429a242deb264b707379646b7/keras/layers/pooling/global_average_pooling1d.py#L93

**Comment by tilakrayal** [comment]
@ehtshamelahiwm,
Could you please provide context/use-case for the issue, why it is needed to change/update the **tf.reduce_sum** to **tf.math.divide_no_nan.**

`tf.math.divide_no_nan` computes a safe divide which returns 0 if y (denominator) is zero whereas `tf.reduce_sum` tries to Compute the sum of elements across dimensions of a tensor. Both are absolutely different in computation.

Thank you!

**Comment by ehtshamelahiwm** [comment]
I meant that the division which is taking place in that step should be a safe division. If the `tf.reduce_sum(mask, axis=steps_axis, keepdims=self.keepdims) ` is zero (which can happen), the global average pooling becomes nan

**Comment by tilakrayal** [comment]
@ehtshamelahiwm,
Could you please provide the test case for the above requested change. So that it will be easy to analyse the modification mentioned. Also using  tf.math.divide_no_nan doesn't have an impact in the code and tf.reduce_sum provides the output which was expected. Thank you!
